Sources/ProjectDescription/TuistConfig.swift

```diff
@@ -20,3 +22,108 @@ public class TuistConfig: Codable {
         dumpIfNeeded(self)
     }
 }
+
+public struct TemplateString: Encodable, Decodable, Equatable {
+    /// Contains a string that can be interpolated with options.
+    let rawString: String
+}
+
+extension TemplateString: ExpressibleByStringLiteral {
+    public init(stringLiteral: String) {
+        rawString = stringLiteral
+    }
+}
+
+extension TemplateString: CustomStringConvertible {
+    public var description: String {
+        return rawString
+    }
+}
+
+extension TemplateString: ExpressibleByStringInterpolation {
+    public init(stringInterpolation: StringInterpolation) {
+        rawString = stringInterpolation.string
+    }
+
+    public struct StringInterpolation: StringInterpolationProtocol {
+        var string: String
+
+        public init(literalCapacity _: Int, interpolationCount _: Int) {
+            string = String()
+        }
+
+        public mutating func appendLiteral(_ literal: String) {
+            string.append(literal)
+        }
+
+        public mutating func appendInterpolation(_ token: TemplateString.Token) {
+            string.append(token.rawValue)
+        }
+    }
+}
+
+extension TemplateString {
+    /// Provides a template for existing project properties.
+    ///
+    /// - projectName: The name of the project.
+    public enum Token: String {
+        case projectName = "${project_name}"
+    }
+}
+
+extension TuistConfig.GenerationOption {
+    enum CodingKeys: String, CodingKey {
+        case generateManifest
+        case xcodeProjectName
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        if container.allKeys.contains(.generateManifest), try container.decodeNil(forKey: .generateManifest) == false {
+            self = .generateManifest
+            return
+        }
+        if container.allKeys.contains(.xcodeProjectName), try container.decodeNil(forKey: .xcodeProjectName) == false {
+            var associatedValues = try container.nestedUnkeyedContainer(forKey: .xcodeProjectName)
+            let associatedValue0 = try associatedValues.decode(TemplateString.self)
+            self = .xcodeProjectName(associatedValue0)
+            return
+        }
+        throw DecodingError.dataCorrupted(.init(codingPath: decoder.codingPath, debugDescription: "Unknown enum case"))
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        switch self {
+        case .generateManifest:
+            _ = container.nestedContainer(keyedBy: CodingKeys.self, forKey: .generateManifest)
+        case let .xcodeProjectName(associatedValue0):
```

Comment:
The name `associatedValue0` doesn't say much about what the variable represents. I'd rename it into something more representative.